Products/ZenUI3/browser/resources/js/zenoss/ExtOverrides.js

```diff
@@ -1098,4 +1098,50 @@ Ext.override(Ext.util.Sorter, {
         }
     });
 
+    Ext.data.Field.override({
+        constructor: function () {
+            var me = this;
+            me.callParent(arguments);
+
+            if (me.type === Ext.data.Types.AUTO) {
+                var customConvert = me.convert;
+
+                me.convert = function (value, record) {
+                    if (Ext.isObject(value) && 'uid' in value) {
+                        value['uid'] = Zenoss.render.link(null, value['uid']);
+                    }
+
+                    if (me.hasCustomConvert) {
+                        return customConvert.apply(me, arguments);
+                    }
+
+                    return value;
+                }
+            }
+        }
+    });
+
+    Ext.data.reader.Reader.override({
+        addVirtualRoot: function (records) {
+            Ext.Array.each(records, function(record, index) {
+                var uid = record.get('uid');
+
+                if (uid) {
+                    record.set({
+                        uid: Zenoss.render.link(null, uid)
+                    });
+                }
+            });
+        },
+
+        extractData: function (root) {
```

Comment:
Override `extractData` of `Ext.data.reader.Reader`:
Deal with the problem with `/cz/` at the stage when the data only come from the server to any store we have intercept returned results from zope server, and before they are populated to store verify if we have there "uid" field, if so add `/cz/` prefix to it;

Reply:
Since all data that are used in trees/grids/combos/forms in ExtJs are handled via store functionality it's good to intercept them in store reader and add missing prefix before we show these data in components. This override doesn't break ExtJs source functionality only extend it.